subprojects/core/src/main/java/org/gradle/api/internal/artifacts/publish/ArchivePublishArtifact.java

```diff
@@ -13,95 +13,21 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
 package org.gradle.api.internal.artifacts.publish;
 
-import org.gradle.api.artifacts.ConfigurablePublishArtifact;
 import org.gradle.api.tasks.bundling.AbstractArchiveTask;
-import org.gradle.util.GUtil;
-
-import java.io.File;
-import java.util.Date;
 
-public class ArchivePublishArtifact extends AbstractPublishArtifact implements ConfigurablePublishArtifact {
-    private String name;
-    private String extension;
-    private String type;
-    private String classifier;
-    private Date date;
-    private File file;
-
-    private AbstractArchiveTask archiveTask;
+public class ArchivePublishArtifact extends AbstractArchivePublishArtifact {
+    private final AbstractArchiveTask archiveTask;
 
     public ArchivePublishArtifact(AbstractArchiveTask archiveTask) {
         super(archiveTask);
         this.archiveTask = archiveTask;
```

Comment:
As Adam mentioned, I think we could merge this class with the `ArchiveProviderPublishArtifact`. We can converge to the common denominator with `Providers.of(...)`.